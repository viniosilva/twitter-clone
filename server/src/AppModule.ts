import { Module } from '@nestjs/common';
import { TerminusModule } from '@nestjs/terminus';
import { LoggerModule } from 'nestjs-pino';
import { httpLogger } from './infra/logger/httpLogger';
import { HealthController } from './presentation/health/HealthController';

@Module({
  imports: [TerminusModule, LoggerModule.forRoot({ pinoHttp: httpLogger })],
  controllers: [HealthController],
})
export default class AppModule {}
