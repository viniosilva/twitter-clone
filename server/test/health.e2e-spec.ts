import { Test, TestingModule } from '@nestjs/testing';
import { INestApplication } from '@nestjs/common';
import request, { Response } from 'supertest';
import AppModule from '../src/AppModule';

describe('HealthController (e2e)', () => {
  const path = '/api/health';
  let app: INestApplication;

  beforeEach(async () => {
    const moduleFixture: TestingModule = await Test.createTestingModule({
      imports: [AppModule],
    }).compile();

    app = moduleFixture.createNestApplication();
    await app.init();
  });

  describe('GET /api/health', () => {
    it('should return successfully', () => {
      return request(app.getHttpServer())
        .get(path)
        .expect(200)
        .expect((res: Response) => {
          expect(res.body).toMatchSnapshot();
        });
    });
  });
});
