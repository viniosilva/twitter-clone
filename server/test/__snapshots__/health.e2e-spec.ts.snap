// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HealthController (e2e) GET /api/health should return successfully 1`] = `"{\\"status\\":\\"ok\\",\\"info\\":{},\\"error\\":{},\\"details\\":{}}"`;
